cmd/juju/application/upgradecharm.go

```diff
@@ -271,6 +273,20 @@ func (c *upgradeCharmCommand) Run(ctx *cmd.Context) error {
 		return errors.Trace(err)
 	}
 
```

Comment:
This is a simpler fix. Other fixes require changing client or external package interfaces (or both). This change ensures that a charm URL with an empty string as a series will not be submitted as a parameter to the server API (the conditions that causes the bug). If a empty string is found when looking up the currently deployed charm (which happens if the running charm was downloaded from a URL without a series) then the application now uses the series found in the application state (which could have come from the command line or otherwise).